README.md

```diff
@@ -155,6 +155,30 @@ pipenv run pytest ./tests
 docker-compose -f docker-compose.testing.yml up && docker-compose -f docker-compose.testing.yml down
 ```
 
+### Migrations
+We are using mongodb as our database, which is a NoSQL, schema-less, document-based DB. However, we are also using `mongoengine` - an
+ODM (Object Document Mapping), which defines a "schema" for each document (like specifying field names or required values). 
+This means that we need a minimalistic migration system to apply the defined "schema" changes, 
+like changing a field name or dropping a collection, if we want to maintain the Application <=> DB integrity.
+
+To create a new migration:
+1. In the `/recommender/migrate` dir:
+2. Create a python module with a name `YYYYMMDDHHSS_migration_name.py` (e.g. `20211126112815_remove_unused_collections.py`)
+3. In this module create a class `MigrationName` (e.g. `RemoveUnusedCollection`) which inherits from `BaseMigration`
+4. Implement `up` (application) and `down` (teardown) methods, by using `self.pymongo_db` ([pymongo](https://pymongo.readthedocs.io/en/stable/), a low-level adapter for mongoDB, connected to our recommender DB instance)
+
+(See existing files in the `/recommender/migrate` dir for more detailed example.)
+
+Migration flask CLI commands:
+- `flask migrate apply` - applies unapplied migrations
+- `flask migrate rollback` - unapplies last applied migration
+- `flask migrate list` - lists all migrations along with their application status
+- `flask migrate check` - checks the integrity of the migrations - if the migration files match the DB migration cache
+- `flask migrate repopopulate` - deletes migration cache and repopulates it with all the migrations defined in `/recommender/migrate` dir.
```

Comment:
`flask migrate repopulate`

Reply:
Shouldn't it also perform all rollbacks on completed migrations?

Reply:
In the case of the integrity check failing (the migrations in the DB cache do not all have their file equivalents (in the recommender/migrations)), there is no point to rollbacking rest of the migrations since the integrity chain has been broken and you cannot undo certain actions (because the migration file is missing)